airbyte-integrations/bases/base-singer/base_singer/source.py

```diff
@@ -68,3 +68,61 @@ def read(
 
         read_cmd = self.read_cmd(logger, config_container.rendered_config_path, selected_singer_catalog_path, state_path)
         return SingerHelper.read(logger, read_cmd)
+
+
+class BaseSingerSource(SingerSource):
+    force_full_refresh = False
+
+    def discover_cmd(self, logger: AirbyteLogger, config_path: str) -> str:
+        return f"{self.tap_cmd} --config {config_path} --discover"
+
+    def read_cmd(self, logger: AirbyteLogger, config_path: str, catalog_path: str, state_path: str = None) -> str:
+        state_path = None if self.force_full_refresh else state_path
+        args = {"--config": config_path, "--catalog": catalog_path, "--state": state_path}
```

Comment:
if force full refresh is true we need to do a few things: 

in read_cmd, we don't want to pass in state
override `discover` [like in here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-github-singer/source_github_singer/source.py#L51) ( while keeping `discover_cmd` since that is run in the SingerSource `read`)

Reply:
ok, so it works now as expected,
the `state` will not be passed as it is None when force_full_refresh is True